compiler/rustc_macros/src/diagnostics/diagnostic_builder.rs

```diff
@@ -249,6 +249,8 @@ impl DiagnosticDeriveBuilder {
         if self.should_generate_set_arg(&field) {
             let diag = &self.diag;
             let ident = field.ident.as_ref().unwrap();
+            // strip `r#` prefix, if present
+            let ident = format_ident!("{}", ident);
```

Comment:
It's a hack, but I haven't found a better way to do this - `proc_macro2::Ident` is very opaque.